.circleci/config.yml

```diff
@@ -15,29 +15,37 @@ parameters:
     default: dps-releases
 
 executors:
-  java-machine:
-    machine:
-      #      TODO: this should be upgraded as soon as a java 16+ machine image is available,
-      #        see https://circleci.com/docs/2.0/configuration-reference/#available-machine-images
-      #      When it is, the kotlinOptions.jvmTarget setting in build.gradle.kts can then be uncommented
-      image: default
-    environment:
-      _JAVA_OPTIONS: -Xmx1024m -XX:ParallelGCThreads=2 -XX:ConcGCThreads=2 -XX:ParallelGCThreads=2 -Djava.util.concurrent.ForkJoinPool.common.parallelism=2 -Dorg.gradle.daemon=false -Dorg.gradle.jvmargs=-XX:+UseContainerSupport -Dkotlin.compiler.execution.strategy=in-process
-    working_directory: ~/app
+  container_dependencies:
+    resource_class: medium+
+    docker:
+      - image: cimg/openjdk:21.0.2
+      - image: "postgis/postgis"
+        environment:
+          - POSTGRES_USER=form-builder
+          - POSTGRES_PASSWORD=form-builder
+          - POSTGRES_DB=form-builder
+      - image: localstack/localstack:3
+        environment:
+          - SERVICES=sns,sqs,s3
+          - DEBUG=${DEBUG- }
+          - DOCKER_HOST=unix:///var/run/docker.sock
+          - DEFAULT_REGION=eu-west-2
+          - ES_PORT_EXTERNAL=4571
+          - DATA_DIR=/tmp/localstack/data
+      - image: "bitnami/redis:7.0.11"
+        environment:
+          - ALLOW_EMPTY_PASSWORD=yes
```

Comment:
Can you not use the hmpps/java_localstack_postgres executor? (here is an example: https://github.com/ministryofjustice/hmpps-prisoner-from-nomis-migration/blob/f12957b0ec8345a29ad115598446643a9d9aa4ee/.circleci/config.yml#L17
)

Reply:
Yes I did try that at first and was talking to Matt Whitaker. I had an issue getting redis started as well using that executor. I looked through the orb project but couldnt find anything suitable to start Redis, its only available as a service in localstack pro.